teps/0084-endtoend-provenance-collection.md

```diff
@@ -237,6 +545,31 @@ via CLI, dashboard or a monitoring system.
 Consider including folks that also work on CLI and dashboard.
 -->
 
+<!-- TODO: Revisit this once event-payload is taken into consideration -->
+
+<!-- TODO: Is there a CLI plugin for chains? -->
+
+To fully take advantage of pipelinerun attestations, pipeline authors should include the
+appropriate type hinting result for Tekton Chains, similarly to how task authors do this today.
+See [Chains Type Hinting](https://tekton.dev/docs/chains/config/#chains-type-hinting) for more
+information.
+
+New configuration options will be introduced to control the behavior for producing and storing
+pipelinerun attestations. These should behave as similar as possible to their taskrun counterparts.
+
+| Key | Description | Supported Values | Default |
+| --- | ----------- | ---------------- | ------- |
+| `artifacts.pipelinerun.format`  | The format to store TaskRun payloads in. | tekton, in-toto | tekton |
+| `artifacts.pipelinerun.storage` | The storage backend to store PipelineRun signatures and attestations in. Multiple backends can be specified with comma-separated list (“tekton,oci”). To disable the PipelineRun artifact input an empty string (""). | tekton, oci, gcs, docdb, grafeas | tekton |
+| `artifacts.pipelinerun.signer`  | The signature backend to sign Taskrun payloads with. | x509, kms | x509 |
```

Comment:
Yah I think that's a good idea! We could start using that and alias the current `artifacts.taskrun` config to map to it. 

I think we should call it something other than`tekton` though. Just because we have `tekton` as a storage type and a format type, I feel like it could be confusing as an artifact type as well. Maybe just `artifacts.run`? Definitely open to suggestions 😄

Reply:
+1 to this conversation. One thing that I keep thinking of is that although Chains is triggered by Tekton artifacts (TaskRun, PipelineRun), it also operates on non-Tekton artifacts (container image, and maven builds (soon?)). Maybe the `artifacts` terminology is becoming a bit misleading.

How about something like `signature.storage`, `signature.format`, `attestation.storage`, etc?